src/azure-cli-core/azure/cli/core/cloud.py

```diff
@@ -477,8 +515,16 @@ def get_active_cloud_name(cli_ctx):
     try:
         return cli_ctx.config.get('cloud', 'name')
     except (configparser.NoOptionError, configparser.NoSectionError):
-        _set_active_cloud(cli_ctx, AZURE_PUBLIC_CLOUD.name)
+        default_cloud_name = get_default_cloud_name()
+        _set_active_cloud(cli_ctx, default_cloud_name)
+        return default_cloud_name
+
+
+def get_default_cloud_name():
+    """ Pick AzureCloud as the default cloud if it is available, otherwise pick the first in the list"""
+    if AZURE_PUBLIC_CLOUD.name in [c.name for c in KNOWN_CLOUDS]:
         return AZURE_PUBLIC_CLOUD.name
+    return KNOWN_CLOUDS[0].name
```

Comment:
Is there a scenario that KNOWN_CLOUDS is None?

Reply:
Added error handling in `get_known_clouds()` to make sure `KNOWN_CLOUDS` is not None or empty.